x-pack/plugins/lens/public/datasources/form_based/operations/definitions/formula/context_variables.tsx

```diff
@@ -0,0 +1,269 @@
+  type: 'time_range',
+  label: 'Time range',
+  description: i18n.translate('xpack.lens.indexPattern.timeRange.documentation.markdown', {
+    defaultMessage: `
+The current time range expressed in milliseconds (ms).
+    `,
+  }),
+  getContextValue: getTimeRangeFromContext,
+  getErrorMessage: getTimeRangeErrorMessages,
+});
+
+export interface NowIndexPatternColumn extends ReferenceBasedIndexPatternColumn {
+  operationType: 'now';
+}
+
+function getNowFromContext({ now }: ContextValues) {
+  return now == null ? Date.now() : +now;
+}
+function getNowErrorMessage() {
+  return undefined;
+}
+
+export const nowOperation = createContextValueBasedOperation<NowIndexPatternColumn>({
+  type: 'now',
+  label: 'Current now',
+  description: i18n.translate('xpack.lens.indexPattern.now.documentation.markdown', {
+    defaultMessage: `
+  The current now moment used in Kibana expressed in milliseconds (ms).
+      `,
+  }),
+  getContextValue: getNowFromContext,
+  getErrorMessage: getNowErrorMessage,
+});
+
+export interface IntervalIndexPatternColumn extends ReferenceBasedIndexPatternColumn {
+  operationType: 'interval';
+}
+
+function getIntervalFromContext(context: ContextValues) {
+  return context.dateRange && context.targetBars
+    ? calcAutoIntervalNear(context.targetBars, getTimeRangeFromContext(context)).asMilliseconds()
+    : 0;
+}
+
+function getIntervalErrorMessages(
+  layer: FormBasedLayer,
+  columnId: string,
+  indexPattern: IndexPattern,
+  dateRange?: DateRange | undefined,
+  operationDefinitionMap?: Record<string, GenericOperationDefinition> | undefined,
+  targetBars?: number
+) {
+  const errors = [];
+  if (!targetBars) {
+    errors.push(
+      i18n.translate('xpack.lens.indexPattern.interval.noTargetBars', {
+        defaultMessage: `Missing "{uiSettingVar}" value`,
+        values: {
+          uiSettingVar: UI_SETTINGS.HISTOGRAM_BAR_TARGET,
+        },
+      })
+    );
+  }
+  if (!dateRange) {
+    errors.push(
+      i18n.translate('xpack.lens.indexPattern.interval.noTimeRange', {
+        defaultMessage: 'The current time range interval is not available',
+      })
+    );
+  }
+  if (
+    !Object.values(layer.columns).some((column) =>
+      isColumnOfType<DateHistogramIndexPatternColumn>('date_histogram', column)
+    )
+  ) {
+    errors.push(
+      i18n.translate('xpack.lens.indexPattern.interval.noDateHistogramColumn', {
+        defaultMessage: 'Cannot compute an interval without a date histogram column configured',
+      })
+    );
+  }
+  return errors.length ? errors : undefined;
+}
+
+export const intervalOperation = createContextValueBasedOperation<IntervalIndexPatternColumn>({
+  type: 'interval',
+  label: 'Date histogram interval',
+  description: i18n.translate('xpack.lens.indexPattern.interval.documentation.markdown', {
+    defaultMessage: `
+  The current date histogram interval bucket expressed in milliseconds (ms).
```

Comment:
How about:

`The specified minimum interval for the date histogram, in milliseconds (ms).`